crates/holochain/tests/tests/countersigning/session_interaction_over_websocket.rs

```diff
@@ -0,0 +1,798 @@
+            CountersigningError::SessionNotFound(alice.cell_id.clone()),
+        ))
+        .into(),
+    );
+    assert_eq!(format!("{:?}", response), format!("{:?}", expected_error));
+
+    // Set up the session and accept it for both agents.
+    let preflight_request: PreflightRequest = call_zome(
+        &alice,
+        &alice_app_tx,
+        "generate_countersigning_preflight_request_fast", // 10 sec timeout
+        &[
+            (alice.cell_id.agent_pubkey().clone(), vec![Role(0)]),
+            (bob.cell_id.agent_pubkey().clone(), vec![]),
+        ],
+    )
+    .await;
+    let alice_acceptance: PreflightRequestAcceptance = call_zome(
+        &alice,
+        &alice_app_tx,
+        "accept_countersigning_preflight_request",
+        &preflight_request,
+    )
+    .await;
+    let alice_response =
+        if let PreflightRequestAcceptance::Accepted(ref response) = alice_acceptance {
+            response
+        } else {
+            unreachable!();
+        };
+    let bob_acceptance: PreflightRequestAcceptance = call_zome(
+        &bob,
+        &bob_app_tx,
+        "accept_countersigning_preflight_request",
+        &preflight_request,
+    )
+    .await;
+    let bob_response = if let PreflightRequestAcceptance::Accepted(ref response) = bob_acceptance {
+        response
+    } else {
+        unreachable!();
+    };
+
+    // Countersigning session state should exist for both agents and be in "Accepted" state.
+    assert_matches!(
+        get_session_state(&alice.cell_id, &alice_app_tx).await,
+        Some(CountersigningSessionState::Accepted(_))
+    );
+    assert_matches!(
+        get_session_state(&bob.cell_id, &bob_app_tx).await,
+        Some(CountersigningSessionState::Accepted(_))
+    );
+
+    // Abandoning a session in a resolvable state should not be possible and return an error.
+    let response: AppResponse = request(
+        AppRequest::AbandonCountersigningSession(Box::new(alice.cell_id.clone())),
+        &alice_app_tx,
+    )
+    .await;
+    let expected_error = AppResponse::Error(
+        ConductorApiError::ConductorError(ConductorError::CountersigningError(
+            CountersigningError::SessionNotUnresolvable(alice.cell_id.clone()),
+        ))
+        .into(),
+    );
+    assert_eq!(format!("{:?}", response), format!("{:?}", expected_error));
+
+    // Publishing a session in a resolvable state should not be possible and return an error.
+    let response: AppResponse = request(
+        AppRequest::PublishCountersigningSession(Box::new(alice.cell_id.clone())),
+        &alice_app_tx,
+    )
+    .await;
+    let expected_error = AppResponse::Error(
+        ConductorApiError::ConductorError(ConductorError::CountersigningError(
+            CountersigningError::SessionNotUnresolvable(alice.cell_id.clone()),
+        ))
+        .into(),
+    );
+    assert_eq!(format!("{:?}", response), format!("{:?}", expected_error));
```

Comment:
Check one more time after trying to abandon and publish that the session is not affected? It should still be `Accepted`. Just because those calls returned errors, it would still be nice to say that the state wasn't affected.